agent/php_mysqli.c

```diff
@@ -430,7 +434,13 @@ static nr_status_t nr_php_mysqli_link_real_connect(
     ADD_IF_STR_SET(argv, argc, metadata->database);
     ADD_IF_INT_SET(argv, argc, metadata->port);
     ADD_IF_STR_SET(argv, argc, metadata->socket);
-    ADD_IF_INT_SET(argv, argc, metadata->flags);
+
+    /*
+     * Avoid defaulting the final $flags argument to null.
+     */
+    if (metadata->flags) {
```

Comment:
won't this miss the case where the flags are actually set to the integer 0?

Reply:
It's okay to not add the flag param if it is zero, since if it's not added, it's zero anyway.

```
mysqli_real_connect(
    mysqli $mysql,
    ?string $hostname = null,
    ?string $username = null,
    #[\SensitiveParameter] ?string $password = null,
    ?string $database = null,
    ?int $port = null,
    ?string $socket = null,
    int $flags = 0
): bool
```